contollers/bookings.ts

```diff
@@ -0,0 +1,90 @@
+import { Request, Response } from "express";
+import { bookingType } from "../utils/types";
+import validator from "email-validator";
+import moment from "moment";
+
+//Use this array as your (in-memory) data store.
+let bookings: bookingType[] = require("../bookings.json");
+
+const validateBook = (booking: bookingType) => {
+  if (
+    !booking.title ||
+    !booking.firstName ||
+    !booking.surname ||
+    !booking.email ||
+    !validator.validate(booking.email) ||
+    !booking.roomId ||
+    !booking.checkInDate ||
+    !booking.checkOutDate ||
+    !moment(booking.checkOutDate).isAfter(booking.checkInDate)
+  ) {
+    return false;
+  }
+  return true;
+};
+export const getAllBookings = (req: Request, res: Response) => {
+  res.send(bookings);
+};
+
+export const getBooking = (req: Request, res: Response) => {
+  const haveBooking = bookings.find(({ id }) => id === +req.params.bookingsId);
+  if (!haveBooking) {
+    return res.status(404).json(`no booking with id ${req.params.bookingsId}`);
+  }
+  return res.send(haveBooking);
+};
+export const searchBookings = (req: Request, res: Response) => {
+  if (!req.query.term && !req.query.date) return;
+
+  if (req.query.term) {
+    const keys = ["firstName", "surname", "email"];
+    const filtered = bookings.filter((booking) =>
+      keys.some((key) =>
+        (booking[key] as string)
+          .toLowerCase()
+          .includes(req.query.term!.toString().toLowerCase())
+      )
+    );
+    return res.send(filtered);
+  }
+  if (moment(req.query.date as string, "YYYY-MM-DD").isValid()) {
```

Comment:
@samhoooo Actually, it might be better to change 'as String' to '<string>' here?